File: go/api/Led.go
package api

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// Led 对应 Led.py。
type Led struct {
	*session.ApiGroup
}

// NewLed 创建 Led API 分组。
func NewLed(s *session.Session) *Led {
	return &Led{ApiGroup: session.NewApiGroup(s)}
}

// Nightmode 对应 led/nightmode。
func (l *Led) Nightmode() (map[string]interface{}, error) {
	return l.S.Get("led/nightmode", nil, "api")
}

// Appctrlled 对应 led/appctrlled。
func (l *Led) Appctrlled() (map[string]interface{}, error) {
	return l.S.Get("led/appctrlled", nil, "api")
}

File: src/api/led.rs
//! Led API group (`api/Led.py`).

use serde_json::Value;

use crate::connection::Connection;
use crate::errors::Result;

use super::get_value;

/// Led API group.
pub struct Led<'a> {
    conn: &'a Connection,
}

impl<'a> Led<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        Led { conn }
    }

    /// `led/nightmode`. LED night mode status.
    pub fn nightmode(&self) -> Result<Value> {
        get_value(self.conn, "api/led/nightmode")
    }

    /// `led/appctrlled`. LED app control status.
    pub fn appctrlled(&self) -> Result<Value> {
        get_value(self.conn, "api/led/appctrlled")
    }
}
